pkg/controllers/nodeclass/status/images.go

```diff
@@ -42,17 +46,32 @@ import (
 
 const (
 	nodeImageReconcilerName = "nodeclass.images"
+
+	// ConfigMap consts
+	maintenanceWindowConfigMapName = "upcoming-maintenance-window"
+	nodeOSMaintenanceWindowChannel = "aksManagedNodeOSUpgradeSchedule"
+	configMapStartTimeFormat       = "%s-start"
+	configMapEndTimeFormat         = "%s-end"
 )
 
 type NodeImageReconciler struct {
-	nodeImageProvider imagefamily.NodeImageProvider
-	cm                *pretty.ChangeMonitor
+	nodeImageProvider            imagefamily.NodeImageProvider
+	inClusterKubernetesInterface kubernetes.Interface
+	systemNamespace              string
+	cm                           *pretty.ChangeMonitor
 }
 
-func NewNodeImageReconciler(provider imagefamily.NodeImageProvider) *NodeImageReconciler {
+func NewNodeImageReconciler(
+	provider imagefamily.NodeImageProvider,
+	inClusterKubernetesInterface kubernetes.Interface,
```

Comment:
I almost wonder if we want this to be a MaintenanceWindow provider? It could abstract the configmap-ness away, which feels good given that nodeclass/status doesn't feel like the right place for the MW logic to live - i.e. fail open or fail closed, etc.

Reply:
Plus it better matches the pattern of passing providers around not passing raw clients?